internal/cephfs/nodeserver.go

```diff
@@ -446,23 +448,37 @@ func (ns *NodeServer) NodePublishVolume(
 	targetPath := req.GetTargetPath()
 	volID := fsutil.VolumeID(req.GetVolumeId())
 
+	volOptions := &store.VolumeOptions{}
+	defer volOptions.Destroy()
+
+	if err := store.ExtractMounter(&volOptions.Mounter, req.GetVolumeContext()); err != nil {
+		return nil, err
+	}
+
+	volMounter, err := mounter.New(volOptions)
+	if err != nil {
+		return nil, err
+	}
+
 	// Considering kubelet make sure the stage and publish operations
 	// are serialized, we dont need any extra locking in nodePublish
 
-	if err := util.CreateMountPoint(targetPath); err != nil {
+	if err = util.CreateMountPoint(targetPath); err != nil {
 		log.ErrorLog(ctx, "failed to create mount point at %s: %v", targetPath, err)
 
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	if err := ns.tryRestoreFuseMountsInNodePublish(
-		ctx,
-		volID,
-		stagingTargetPath,
-		targetPath,
-		req.GetVolumeContext(),
-	); err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to try to restore FUSE mounts: %v", err)
+	if _, ok := volMounter.(*mounter.FuseMounter); ok {
+		if err = ns.tryRestoreFuseMountsInNodePublish(
```

Comment:
as the check is added outside, we can remove few things inside the tryRestoreFuseMountsInNodePublish function, please check.